pkg/agent/config.go

```diff
@@ -302,7 +302,7 @@ func InitAgentCmdFlags(c *cobra.Command, cfg *AgentCmdFlags) {
 		&cfg.DisableCompression,
 		"disable-compression",
 		false,
-		"Disables GZIP compression when uploading the data. Useful for debugging purposes or when an intermediate proxy doesn't like compressed data.",
+		"Deprecated. No longer has an effect.",
```

Comment:
Optional. Consider using the `pflag.MarkDeprecated` method instead, which will hide the flag from the --help and therefore from the documented CLI help.

Reply:
I've tried this:

```
c.PersistentFlags().MarkDeprecated("disable-compression", "no longer has an effect")
```

But it uses logrus' logger, so it doesn't show as JSON (first line of the below snippet):

```
Flag --disable-compression has been deprecated, no longer has an effect
I1122 18:09:24.792410   25556 run.go:59] "Starting" logger="Run" version="development" commit=""
I1122 18:09:24.793470   25556 config.go:404] "Using the Venafi Cloud Key Pair Service Account auth mode since --client-id and --private-key-path were specified." logger="Run"
I1122 18:09:24.793480   25556 config.go:540] "Using period from config" logger="Run" period="5m0s"
I1122 18:09:24.793486   25556 config.go:592] "The flag --disable-compression has been deprecated an no longer has any effect." logger="Run"
```

Is that OK?

Reply:
Use MarkHidden instead?

Reply:
I've decided to use `MarkDeprecated` instead. I don't think this plain text line is a big line, and I can still try to figure how to make it better later.